Sources/StreamChat/Utils/EventBatcher.swift

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// The type that does events batching.
+protocol EventBatcher {
+    typealias Batch = [Event]
+    
+    /// The current batch of events.
+    var currentBatch: Batch { get }
+    
+    /// Creates new batch processor.
+    init(period: TimeInterval, handler: @escaping (Batch) -> Void)
+    
+    /// Adds the item to the current batch of events. If it's the first event also schedules batch processing
+    /// that will happen when `period` has passed.
+    ///
+    /// - Parameter event: The event to add to the current batch.
+    func append(_ event: Event)
+    
+    /// Ignores `period` and passes the current batch of events to handler as soon as possible.
+    func processImmidiately()
+}
+
+extension Batcher: EventBatcher where Item == Event {}
+
+final class Batcher<Item> {
```

Comment:
Do we really need the protocol, and the generic class?

Reply:
Good question 🙂 I was thinking like that: we need a protocol to be able to inject a mock that passes events through without batching. But events it's not the only thing we might wanna batch so would be nice if it can be reused for any other item type